src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/DoublyLinkedList.cs

```diff
@@ -14,7 +14,7 @@ namespace System.Text.RegularExpressions.Symbolic
     /// <see cref="AddLast(DoublyLinkedList{T})"/> operation is made use of in the <see cref="RegexNodeConverter.ConvertToSymbolicRegexNode(RegexNode)"/> method
     /// where it maintains linear construction time in terms of the overall number of AST nodes in a given <see cref="RegexNode"/> input.
     /// </remarks>
-    internal sealed class DoublyLinkedList<T> where T : notnull
+    internal sealed class DoublyLinkedList<T> : IEnumerable<T> where T : notnull
```

Comment:
A remark that the `IEnumerable` is implemented in reverse might help avoid future confusion here.